service/src/main/java/bio/terra/cbas/runsets/monitoring/SmartRunSetsPoller.java

```diff
@@ -99,6 +102,30 @@ private RunSet updateRunSet(RunSet rs, OffsetDateTime limitedEndTime) {
       smartRunsPoller.updateRuns(updateableRuns, Optional.of(runPollUpdateEndTime));
 
       StatusAndCounts newStatusAndCounts = newStatusAndErrorCounts(rs);
+
+      if (rs.status() == CbasRunSetStatus.CANCELING) {
+        abortManager.abortRunSet(rs.runSetId());
```

Comment:
I think calling `abortRunSet` here might be pre-mature. If all the Runs in the RunSet are in `Canceled` state then we don't need to call this function here. I think this should be called in the `else` section of `if` block on L119.

Reply:
One thing to also note is that in `abortRunSet` method on L41 the status of RunSet is set to Canceling. This should only happen the first time when a request for aborting a RunSet comes in. We shouldn't reset it when its called from the SmartRunSetPoller as we will loose the original timestamp of when the abort request was made.

Reply:
I don't think that I'm setting the runSet to cancelling again in the `SmartRunSetPoller`. I think I'm only setting the run set to a cancelling state when the request comes in. Is that right? Should I be looking somewhere else?

Reply:
I think [this L40](https://github.com/DataBiosphere/cbas/blob/5f3aca90a9e05a1c4a0ce1dc04b2a900f95dc4d2/service/src/main/java/bio/terra/cbas/runsets/monitoring/RunSetAbortManager.java#L40) inside the `abortManager.abortRunSet()`  sets the RunSet to `Canceling` state. When the abort request comes in, the RunSet state should be changed to `Canceling`. But in the SmartRunSetPoller, if the RunSet still has some runs to be cancelled, it calls the `abortManager.abortRunSet()` again which would set the RunSet state and timestamp to Cancelling again. Is my understanding of the code right?

Reply:
Ah, my mistake. So I should make that check conditional, so it's not constantly being set to a 'CANCELING` state. Sorry for my confusion.